src/puppetlabs/puppetdb/cli/export.clj

```diff
@@ -275,14 +175,31 @@
            construct-base-url
            utils/validate-cli-base-url!)))))
 
+(defn export!
+  [outfile nodes-data]
+  (with-open [tar-writer (archive/tarball-writer outfile)]
+    (let [metadata (export-metadata (now))]
+      (log/info (:msg metadata))
+      (utils/add-tar-entry tar-writer metadata))
+    (doseq [tar-item nodes-data]
+      (log/info (:msg tar-item))
+      (utils/add-tar-entry tar-writer tar-item))))
+
+(pls/defn-validated trigger-export-via-http!
+  [base-url :- utils/base-url-schema
+   filename :- s/Str]
+  (clojure.java.io/copy
+   (:body
+    (client/get
+     (str (utils/base-url->str base-url)
+          (format "/export?archive=%s" filename))
+     {:accept :octet-stream
+      :as :stream}))
+   (clojure.java.io/file filename)))
+
 (defn -main
   [& args]
-  (let [{:keys [outfile base-url] :as opts} (validate-cli! args)
-        nodes (get-nodes base-url)]
-    (with-open [tar-writer (archive/tarball-writer outfile)]
-      (utils/add-tar-entry tar-writer (export-metadata))
-      (doseq [node nodes
-              :let [node-data (get-node-data base-url node)]]
-        (doseq [{:keys [msg] :as tar-item} (mapcat node-data [:catalog :reports :facts])]
-          (println msg)
-          (utils/add-tar-entry tar-writer tar-item))))))
+  (let [{:keys [outfile base-url]} (validate-cli! args)]
+    (println (str "Triggering export to " outfile "..."))
+    (trigger-export-via-http! base-url outfile)
+    (println (str "Finished export to " outfile "."))))
```

Comment:
might be redundant with the other logging, but it could be useful to print timestamps in these lines